File: SeriesGuide/src/main/java/com/battlelancer/seriesguide/ui/shows/CalendarQuery.java
package com.battlelancer.seriesguide.ui.shows;

import com.battlelancer.seriesguide.provider.SeriesGuideContract;
import com.battlelancer.seriesguide.provider.SeriesGuideDatabase;

public interface CalendarQuery {

    String[] PROJECTION = new String[] {
            SeriesGuideDatabase.Tables.EPISODES + "." + SeriesGuideContract.Episodes._ID,
            SeriesGuideContract.Episodes.TITLE,
            SeriesGuideContract.Episodes.NUMBER,
            SeriesGuideContract.Episodes.SEASON,
            SeriesGuideContract.Episodes.FIRSTAIREDMS,
            SeriesGuideContract.Episodes.WATCHED,
            SeriesGuideContract.Episodes.COLLECTED,
            SeriesGuideContract.Shows.REF_SHOW_ID,
            SeriesGuideContract.Shows.TITLE,
            SeriesGuideContract.Shows.NETWORK,
            SeriesGuideContract.Shows.POSTER
    };

    String QUERY_UPCOMING = SeriesGuideContract.Episodes.FIRSTAIREDMS + ">=? AND "
            + SeriesGuideContract.Episodes.FIRSTAIREDMS
            + "<? AND " + SeriesGuideContract.Shows.SELECTION_NO_HIDDEN;

    String QUERY_RECENT = SeriesGuideContract.Episodes.SELECTION_HAS_RELEASE_DATE + " AND "
            + SeriesGuideContract.Episodes.FIRSTAIREDMS + "<? AND "
            + SeriesGuideContract.Episodes.FIRSTAIREDMS + ">? AND "
            + SeriesGuideContract.Shows.SELECTION_NO_HIDDEN;

    String SORTING_UPCOMING = SeriesGuideContract.Episodes.FIRSTAIREDMS + " ASC,"
            + SeriesGuideContract.Shows.SORT_TITLE + ","
            + SeriesGuideContract.Episodes.NUMBER + " ASC";

    String SORTING_RECENT = SeriesGuideContract.Episodes.FIRSTAIREDMS + " DESC,"
            + SeriesGuideContract.Shows.SORT_TITLE + ","
            + SeriesGuideContract.Episodes.NUMBER + " DESC";

    int _ID = 0;
    int TITLE = 1;
    int NUMBER = 2;
    int SEASON = 3;
    int RELEASE_TIME_MS = 4;
    int WATCHED = 5;
    int COLLECTED = 6;
    int SHOW_ID = 7;
    int SHOW_TITLE = 8;
    int SHOW_NETWORK = 9;
    int SHOW_POSTER_PATH = 10;
}

File: app/src/main/java/com/battlelancer/seriesguide/ui/shows/CalendarQuery.kt
package com.battlelancer.seriesguide.ui.shows

import com.battlelancer.seriesguide.provider.SeriesGuideContract.Episodes
import com.battlelancer.seriesguide.provider.SeriesGuideContract.Shows
import com.battlelancer.seriesguide.provider.SeriesGuideDatabase

interface CalendarQuery {
    companion object {

        val PROJECTION = arrayOf(
            SeriesGuideDatabase.Tables.EPISODES + "." + Episodes._ID,
            Episodes.TITLE,
            Episodes.NUMBER,
            Episodes.SEASON,
            Episodes.FIRSTAIREDMS,
            Episodes.WATCHED,
            Episodes.COLLECTED,
            Shows.REF_SHOW_ID,
            Shows.TITLE,
            Shows.NETWORK,
            Shows.POSTER
        )

        const val QUERY_UPCOMING = "${Episodes.FIRSTAIREDMS}>=? AND ${Episodes.FIRSTAIREDMS}<? " +
                "AND ${Shows.SELECTION_NO_HIDDEN}"

        const val QUERY_RECENT = "${Episodes.SELECTION_HAS_RELEASE_DATE} " +
                "AND ${Episodes.FIRSTAIREDMS}<? AND ${Episodes.FIRSTAIREDMS}>? " +
                "AND ${Shows.SELECTION_NO_HIDDEN}"

        const val SORTING_UPCOMING = "${Episodes.FIRSTAIREDMS} ASC," +
                "${Shows.SORT_TITLE},${Episodes.NUMBER} ASC"

        const val SORTING_RECENT = "${Episodes.FIRSTAIREDMS} DESC," +
                "${Shows.SORT_TITLE},${Episodes.NUMBER} DESC"

        const val EPISODE_TVDB_ID = 0
        const val TITLE = 1
        const val NUMBER = 2
        const val SEASON = 3
        const val RELEASE_TIME_MS = 4
        const val WATCHED = 5
        const val COLLECTED = 6
        const val SHOW_ID = 7
        const val SHOW_TITLE = 8
        const val SHOW_NETWORK = 9
        const val SHOW_POSTER_PATH = 10
    }
}
